### ISSUE TITLE
Command terminated abnormally error

### ISSUE THREAD

**Issue by pcantalupo** [bbuchfink/diamond#5]
Hello,

I'm running Diamond on a Mac OS X 10.6 snow leopard. I was able to successfully align 1 sequence (https://gist.github.com/pcantalupo/5899d47212e73ec02f35) against the NR database but when I run 100 sequences (https://gist.github.com/pcantalupo/77c59a00a423783c34b3), diamond exits with an error of 1 and outputs "Command terminated abnormally". This happens approx 20 minutes after starting diamond. Here is my command line:

```
diamond blastx -d ~/local/usr/local/ncbi/blast/db/nr -q seqs100.fa -a diamond.seqs100 -t diamond_tmp
```

My system has 8 hyperthreaded cores and 32 GB ram. The NR.dmnd file size is 22.1 GB. I created the dmnd file with NR.faa from NR downloaded in late 2013.

Thank you for your help,

Paul

**Comment by bbuchfink** [comment]
Hi,
could you run the program with option -v and show me all of the console output?
Benjamin

**Comment by pcantalupo** [comment]
Here it is. It ends with a Seg fault

```
$ diamond blastx -d ~/local/usr/local/ncbi/blast/db/nr -q seqs100.fa -a diamond.seqs100 -t diamond_tmp -v           diamond v0.7.8.57
#Threads = 16
Scoring matrix = blosum62
Lambda = 0.267
K = 0.041
Gap open penalty = 11
Gap extension penalty = 1
Seg masking = 1
SSSE3 enabled.
Opening the database... [0.0s]
Reference = /x/y/local/usr/local/ncbi/blast/db/nr.dmnd
Sequences = 34305433
Letters = 12011942272
Block size = 2000000000
Opening the input file... [0.0s]
Opening the output file... [0.0s]
Loading query sequences... [0.0s]
Sequences = 600, letters = 172564
Running complexity filter... [0.0s]
Building query histograms... [0.0s]
Allocating buffers... [0.0s]
Loading reference sequences... [33.4s]
Allocating buffers... [0.0s]
Initializing temporary storage... [0.1s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 0.
Building reference index... [13.2s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 1.
Building reference index... [13.4s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 2.
Building reference index... [13.2s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 3.
Building reference index... [12.7s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 1, index chunk 0.
Building reference index... [12.7s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 1, index chunk 1.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 1, index chunk 2.
Building reference index... [13.2s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 1, index chunk 3.
Building reference index... [12.6s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 2, index chunk 0.
Building reference index... [12.5s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 2, index chunk 1.
Building reference index... [12.9s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 2, index chunk 2.
Building reference index... [13.0s]
Building query index... [0.0s]
Searching alignments... [0.4s]
Processing query chunk 0, reference chunk 0, shape 2, index chunk 3.
Building reference index... [12.6s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 3, index chunk 0.
Building reference index... [12.4s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 3, index chunk 1.
Building reference index... [12.8s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 3, index chunk 2.
Building reference index... [12.8s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 0, shape 3, index chunk 3.
Building reference index... [12.4s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Closing temporary storage... [0.0s]
Deallocating buffers... [0.5s]
Opening temporary output file... [0.0s]
Computing alignments... [0.2s]
Closing temporary output file... [0.0s]
Deallocating reference... [0.6s]
Loading reference sequences... [43.5s]
Allocating buffers... [0.0s]
Initializing temporary storage... [0.0s]
Processing query chunk 0, reference chunk 1, shape 0, index chunk 0.
Building reference index... [13.2s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 0, index chunk 1.
Building reference index... [13.2s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 0, index chunk 2.
Building reference index... [13.2s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 0, index chunk 3.
Building reference index... [12.7s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 1, index chunk 0.
Building reference index... [12.7s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 1, index chunk 1.
Building reference index... [13.0s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 1, index chunk 2.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 1, index chunk 3.
Building reference index... [12.6s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 2, index chunk 0.
Building reference index... [12.5s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 2, index chunk 1.
Building reference index... [12.9s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 2, index chunk 2.
Building reference index... [12.9s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 2, index chunk 3.
Building reference index... [12.5s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 3, index chunk 0.
Building reference index... [12.4s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 3, index chunk 1.
Building reference index... [12.8s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 3, index chunk 2.
Building reference index... [13.0s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 1, shape 3, index chunk 3.
Building reference index... [12.4s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Closing temporary storage... [0.0s]
Deallocating buffers... [0.5s]
Opening temporary output file... [0.0s]
Computing alignments... [0.1s]
Closing temporary output file... [0.0s]
Deallocating reference... [0.4s]
Loading reference sequences... [42.6s]
Allocating buffers... [0.0s]
Initializing temporary storage... [0.0s]
Processing query chunk 0, reference chunk 2, shape 0, index chunk 0.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 0, index chunk 1.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 0, index chunk 2.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 0, index chunk 3.
Building reference index... [12.9s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 1, index chunk 0.
Building reference index... [12.6s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 1, index chunk 1.
Building reference index... [13.0s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 1, index chunk 2.
Building reference index... [13.2s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 1, index chunk 3.
Building reference index... [12.5s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 2, index chunk 0.
Building reference index... [12.5s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 2, index chunk 1.
Building reference index... [13.0s]
Building query index... [0.0s]
Searching alignments... [0.4s]
Processing query chunk 0, reference chunk 2, shape 2, index chunk 2.
Building reference index... [12.9s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 2, index chunk 3.
Building reference index... [12.5s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 3, index chunk 0.
Building reference index... [12.6s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 3, index chunk 1.
Building reference index... [12.9s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 3, index chunk 2.
Building reference index... [12.8s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 2, shape 3, index chunk 3.
Building reference index... [12.3s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Closing temporary storage... [0.0s]
Deallocating buffers... [0.5s]
Opening temporary output file... [0.0s]
Computing alignments... [0.1s]
Closing temporary output file... [0.0s]
Deallocating reference... [0.4s]
Loading reference sequences... [50.3s]
Allocating buffers... [0.0s]
Initializing temporary storage... [0.1s]
Processing query chunk 0, reference chunk 3, shape 0, index chunk 0.
Building reference index... [13.2s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 0, index chunk 1.
Building reference index... [13.3s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 0, index chunk 2.
Building reference index... [13.3s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 0, index chunk 3.
Building reference index... [12.7s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 1, index chunk 0.
Building reference index... [12.8s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 1, index chunk 1.
Building reference index... [13.0s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 1, index chunk 2.
Building reference index... [13.3s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 1, index chunk 3.
Building reference index... [12.6s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 2, index chunk 0.
Building reference index... [12.5s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 2, index chunk 1.
Building reference index... [12.9s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 2, index chunk 2.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 2, index chunk 3.
Building reference index... [12.6s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 3, index chunk 0.
Building reference index... [12.4s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 3, index chunk 1.
Building reference index... [12.8s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 3, index chunk 2.
Building reference index... [12.8s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 3, shape 3, index chunk 3.
Building reference index... [12.4s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Closing temporary storage... [0.0s]
Deallocating buffers... [0.5s]
Opening temporary output file... [0.0s]
Computing alignments... [0.1s]
Closing temporary output file... [0.0s]
Deallocating reference... [0.5s]
Loading reference sequences... [42.9s]
Allocating buffers... [0.0s]
Initializing temporary storage... [0.0s]
Processing query chunk 0, reference chunk 4, shape 0, index chunk 0.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 4, shape 0, index chunk 1.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 4, shape 0, index chunk 2.
Building reference index... [13.1s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 4, shape 0, index chunk 3.
Building reference index... [12.8s]
Building query index... [0.0s]
Searching alignments... [0.5s]
Processing query chunk 0, reference chunk 4, shape 1, index chunk 0.
Building reference index... [12.7s]
Building query index... [0.0s]
Searching alignments... Segmentation fault
```

**Comment by bbuchfink** [comment]
Sorry but I could not reproduce the problem so far, I will have to keep looking into it.

**Comment by pcantalupo** [comment]
Is it because I only have 32 GB RAM? Are there any options that I should set to limit amount of memory?

Thank you,

**Comment by bbuchfink** [comment]
No, 32 GB should be plenty enough.

**Comment by pcantalupo** [comment]
any possibility that you could release a Mac OS X binary? Thank you, Paul

**Comment by bbuchfink** [comment]
Certainly, but that would probably not fix your problem. It is not a compiler error.

**Comment by bbuchfink** [comment]
I ran the same thing on a MAC myself but didn't get the error. This makes it hard to track down. I'll try and see if I can find another MAC with 10.6.

**Comment by pcantalupo** [comment]
Sorry but I thought of something else (I'm really trying to get this to work on my machine since I need a fast blastx).

Could it be something to do with my nr.dmnd file? I created it by extracting protein sequences from the nr database and then running diamond makedb:
```
blastdbcmd -entry 'all' -db nr > nr.faa
diamond makedb --in nr.faa -d nr
```

I looked in my nr.faa file and there are some really, really long fasta header lines. See https://gist.github.com/pcantalupo/b1a229ad6ea06fd9511e for the first 25 lines of my nr.faa. Could this be a problem?
